eclipse-theia/theia-ide
# fix: remove env variable workaround to define configuration area

#### What it does
This Pull Request removes the workaround used to defined configuration area for Theia IDE. It now relies on the `configurationFolder` backend configuration property. We can then remove the product contribution `TheiaIDEEnvVariableServer`. This was overriding the default environment variable server, and so some part of its code.
Also, there was a slight different behavior between electron and browser version, as electron was adding the theia-ide folder to the `THEIA_PLUGINS` variables on the contrary to the browser version, which makes now no sense. This slight difference is now removed.

fixes #393

contributed on behalf of STMicroelectronics

#### How to test
Launch theia ide browser and electron versions with patch. Check that `.theia-ide` is the configuration folder (it can be deleted prior to the launch). Configuration folder is also logged at startup.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
